apps/fz_http/lib/fz_http_web/controllers/auth_controller.ex

```diff
@@ -50,6 +50,16 @@ defmodule FzHttpWeb.AuthController do
     end
   end
 
+  def callback(conn, %{"provider" => "saml"}) do
+    key = {idp, _} = get_session(conn, "samly_assertion_key")
+    assertion = %Samly.Assertion{} = Samly.State.get_assertion(conn, key)
+
+    with {:ok, user} <-
+           UserFromAuth.find_or_create(:saml, %{"email" => assertion.subject.name, "sub" => ""}) do
```

Comment:
Hm, this will take on the `auto_create_oidc_users` setting. I think we may need to refactor that config so that `auto_create_users` can be enabled on a per-provider basis for both OIDC or SAML. Thoughts on that approach?

Reply:
Yes. It's only done this way to get it working for further exploration.

Reply:
Lots of temporary code will be cleaned up after all the exploratory work is done.